logic_test/set_time_zone: fix DST related test flake

otan opened the pull request:
Tokyo has no DST, so that will prevent this class of errors for now.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/72504)
<!-- Reviewable:end -->

otan committed:
logic_test/set_time_zone: fix DST related test flake

Tokyo has no DST, so that will prevent this class of errors for now.

Release note: None

ajwerner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

`America/Phoenix` amusingly also doesn't do DST anymore :) 

https://en.wikipedia.org/wiki/Time_in_Arizona

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72504#-:-MnwhWdH0UDQekDIJ6_6:b74nt5e)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @RichardJCai)



<!-- Sent from Reviewable.io -->

otan commented:
arizona is so weird though with time zones! i remember driving through and having the time change underneath me several times.

anyway, TFTR! 

bors r=ajwerner
